CHANGELOG.md

```diff
@@ -11,6 +11,10 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
 
 ### Fixed
 
+- Fixed Deep expression matching and metavariables interaction. Semgrep will
+  not stop anymore at the first match and will enumarate all possible matchings
+  if a metavariable is used in a deep expression pattern
+  (e.g., `<... $X ...>`). This can introduce some performance regressions.
```

Comment:
If the old behavior is preferred by some we could always make it optional.

Reply:
true, but I think the old behavior is just wrong. If $X was bounded before, then it's the same, because the first match will fail and force to go deeper, but otherwise it will not explore all possibilities.